### ISSUE TITLE
How should I do if I forward a layer twice but only backpropagate once?

### ISSUE THREAD

**Issue by bachml** [pytorch/pytorch#51215]
Hey guys,

I have a question like this:

<img width="1022" alt="截屏2021-01-28 上午1 23 08" src="https://user-images.githubusercontent.com/9403356/106029202-d1f39900-6107-11eb-9595-80f4083cee90.png">

In this case, I want to reuse the layerA in the phrase 3. LayerA in phrase1 and LayerA in phrase3 has the same weight(layerA in phrase 3 is LayerA in phrase1's copy), but I don't want phrase3 to accumulate gradients(in order word, layerA only update its weight in phrase1, and freeze in phrase3).

Here is the problem: If I set require_grad=False for Layer A, then it won't learn anything even in phrase1. If I set with torch.no_grad(), phrase 3 won't backpropagte loss to phrase2.

How should I do in this situation? thanks!

**Comment by ailzhang** [comment]
Per instruction in the issue template, please redirect all questions to PyTorch forum instead. Thanks!
